prowler/providers/aws/services/rds/rds_service.py

```diff
@@ -14,7 +14,7 @@ def __init__(self, provider):
         # Call AWSService's __init__
         super().__init__(__class__.__name__, provider)
         self.db_instances = []
-        self.db_clusters = {}
+        self.db_clusters = []
```

Comment:
This needs to be kept as a dict using the cluster ARN as key.

Reply:
Changed this to a list so that it would match self.db_instances and could use similar parameter group code. I can switch this back to a dict if required or if it would cause breaking changes.

Reply:
We must use maps to store objects using the ARN as key for faster retrievals.

Reply:
Ahh I understand. Im going to close this PR and redo it entirely. Thank you @jfagoagas for the guidance.